source/common/http/http1/conn_pool.cc

```diff
@@ -30,6 +30,10 @@ ConnPoolImpl::ConnPoolImpl(Event::Dispatcher& dispatcher, Upstream::HostConstSha
       upstream_ready_timer_(dispatcher_.createTimer([this]() { onUpstreamReady(); })) {}
 
 ConnPoolImpl::~ConnPoolImpl() {
+  while (!delayed_clients_.empty()) {
+    delayed_clients_.front()->codec_client_->close();
```

Comment:
does close() remove the client form the delayed_clients_ list?  otherwise looks like an infinite loop if the list is non-empty

Reply:
oh, front() pops off the head of the list

Reply:
This is fine (see loops for other `ActiveClientPtr` lists below).

Reply:
close() will move itself out of the list. I learn it from crash log